### PR TITLE
[SPARK-45070][SQL][DOCS] Describe the binary and datetime formats of `to_char`/`to_varchar`

### PR DESCRIPTION
**Pull request by MaxGekk**
### What changes were proposed in this pull request?
In the PR, I propose to document the recent changes related to the `format` of the `to_char`/`to_varchar` functions:
1. binary formats added by https://github.com/apache/spark/pull/42632
2. datetime formats introduced by https://github.com/apache/spark/pull/42534

### Why are the changes needed?
To inform users about recent changes.

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
By CI.

### Was this patch authored or co-authored using generative AI tooling?
No.

### CODE REVIEW

File: connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/functions.scala [hunk 1]
```diff
@@ -4311,11 +4320,20 @@ object functions {
...
+   *     <li>'base64': a base 64 string.</li>
+   *     <li>'hex': a string in the hexadecimal format.</li>
+   *     <li>'utf-8': the input binary is decoded to UTF-8 string.</li>
+   *   </ul>
    *
    * @group string_funcs
    * @since 3.5.0
    */
   def to_varchar(e: Column, format: Column): Column = Column.fn("to_varchar", e, format)
+  // scalastyle:on line.size.limit
```
Comment (cloud-fan) [inline connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/functions.scala:4336]: shall we put it before the `def to_varchar`? It's better to limit the code block that skips style check

File: sql/core/src/main/scala/org/apache/spark/sql/functions.scala [hunk 2]
```diff
@@ -4446,10 +4454,18 @@ object functions {
...
+   *  If `e` is a binary, it is converted to a string in one of the formats:
+   *     'base64': a base 64 string.
+   *     'hex': a string in the hexadecimal format.
+   *     'utf-8': the input binary is decoded to UTF-8 string.
+   *
    * @group string_funcs
    * @since 3.5.0
    */
   def to_varchar(e: Column, format: Column): Column = call_function("to_varchar", e, format)
+  // scalastyle:on line.size.limit
```
Comment (cloud-fan) [inline sql/core/src/main/scala/org/apache/spark/sql/functions.scala:4468]: ditto

**Review by cloud-fan — APPROVED** [review]

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM.

File: connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/functions.scala [hunk 3]
```diff
@@ -4284,12 +4285,22 @@ object functions {
...
    *   end of the format string; specifies that the result string will be wrapped by angle
    *   brackets if the input value is negative.</li> </ul>
+   *   If `e` is a datetime, `format` shall be a valid datetime pattern, see
+   *   <a href="https://spark.apache.org/docs/latest/sql-ref-datetime-pattern.html">Datetime Patterns</a>.
+   *   If `e` is a binary, it is converted to a string in one of the formats:
+   *   <ul>
+   *     <li>'base64': a base 64 string.</li>
+   *     <li>'hex': a string in the hexadecimal format.</li>
+   *     <li>'utf-8': the input binary is decoded to UTF-8 string.</li>
+   *   </ul>
```
Comment (dongjoon-hyun) [inline connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/functions.scala:4295]: To @MaxGekk , it seems our Linter CI doesn't allow this kind of formatting in the comment.

- https://github.com/MaxGekk/spark/actions/runs/6076537960/job/16484751098

```
The scalafmt check failed on connector/connect at following occurrences:

Requires formatting: functions.scala

Before submitting your change, please make sure to format your code using the following command:
./build/mvn -Pscala-2.12 scalafmt:format -Dscalafmt.skip=false -Dscalafmt.validateOnly=false -Dscalafmt.changedOnly=false -pl connector/connect/common -pl connector/connect/server -pl connector/connect/client/jvm
Error: Process completed with exit code 1.
```

The recommended style is something like the following. Please fix the style by run the above command line.
```
-   *   brackets if the input value is negative.</li> </ul>
-   *   If `e` is a datetime, `format` shall be a valid datetime pattern, see
-   *   <a href="https://spark.apache.org/docs/latest/sql-ref-datetime-pattern.html">Datetime Patterns</a>.
-   *   If `e` is a binary, it is converted to a string in one of the formats:
-   *   <ul>
-   *     <li>'base64': a base 64 string.</li>
-   *     <li>'hex': a string in the hexadecimal format.</li>
-   *     <li>'utf-8': the input binary is decoded to UTF-8 string.</li>
-   *   </ul>
+   *   brackets if the input value is negative.</li> </ul> If `e` is a datetime, `format` shall be
+   *   a valid datetime pattern, see <a
+   *   href="https://spark.apache.org/docs/latest/sql-ref-datetime-pattern.html">Datetime
+   *   Patterns</a>. If `e` is a binary, it is converted to a string in one of the formats: <ul>
+   *   <li>'base64': a base 64 string.</li> <li>'hex': a string in the hexadecimal format.</li>
+   *   <li>'utf-8': the input binary is decoded to UTF-8 string.</li> </ul>
```

Comment (MaxGekk) [reply connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/functions.scala]: Thank you, @dongjoon-hyun

**Comment by MaxGekk** [comment]
Merging to master. Thank you, @dongjoon-hyun and @cloud-fan for review.

**Pull request closed without merging**